### PR TITLE
Correct fill attributes when scrolling and erasing

### LINKED ISSUE #2553: Commands that erase the screen should clear the rendition attributes
# Environment

```none
Windows build number: Version 10.0.18362.239
```

# Steps to reproduce

Open a WSL conhost shell, and execute the following command:

    printf "\e[41;93;4;7m\e[2J\e[m"

This sets the text color to bright yellow on red, and enables the reverse video and underline attributes. It then clears the screen with an `ED` (erase in display) sequence.

# Expected behavior

According to [the VT510 reference](https://vt100.net/docs/vt510-rm/ED.html), this should erase the lines "with all visual character attributes cleared". The DEC STD 070 manual says the same sort of thing: "the empty character (imaged as SPACE), the empty rendition, and the empty character attribute." Rendition refers to attributes like underline, reverse video, etc.

So when clearing the screen, I would expect the underline and reverse attributes not to be applied, so it should just fill the screen with the red background color (DEC's VT terminals never had color, but that's a requirement of the [background color erase](https://invisible-island.net/ncurses/ncurses.faq.html#bce_mismatches) feature).

This is what the result looks like in XTerm:

image

# Actual behavior

In the Windows console, though, the underline and reverse video attributes _are_ applied, so the screen is filled with the yellow foreground color rather than the red background color, and every character on the screen also has a red underline.

image

# More information

Note that this is an issue for a number of escape sequences. There are those that erase areas of the screen directly (`ED`, `EL`, `ECH`), and also some that reveal or delete areas of the screen as a result of scrolling (`SU`, `SD`, `RI`, `IL`, `DL`, `ICH`, `DCH`). If we're going to match the XTerm behaviour, then all of them should be filling the erased area with just the background color, and with none of the rendition attributes set.

It might seem like this could easily be fixed by clearing the `META_ATTRS` of the fill attribute in the calls to [`ScrollConsoleScreenBufferWImpl`](https://github.com/microsoft/terminal/blob/0e6f290806ef066e8ef29c0a933c83a0007eac98/src/host/getset.cpp#L811-L854) and [`FillConsoleOutputAttributeImpl`](https://github.com/microsoft/terminal/blob/9b92986b49bed8cc41fde4d6ef080921c41e6d9e/src/host/_output.cpp#L195-L253). However, that would prevent those methods working correctly with RGB colors, since they rely on [a hack](https://github.com/microsoft/terminal/blob/9b92986b49bed8cc41fde4d6ef080921c41e6d9e/src/host/_output.cpp#L223-L238) that expects the fill attribute to be an exact match for the current text attribute (and that already has problems - see issue #832).

My proposal (not yet tested), is we replace the current hack with a much simpler hack, that just checks for a special case attribute value of `0xFFFF`, which we declare to be the "background erase" color. When passed that value, the fill and scroll methods would simply convert that to the current value of `SCREEN_INFORMATION::GetAttributes` but with the meta attributes cleared. Its easy for the calling code to set, and doesn't require an expensive `GenerateLegacyAttributes` calculation on either side of the call.

### PR DESCRIPTION
**Pull request by j4james**
## Summary of the Pull Request

Operations that erase areas of the screen are typically meant to do so using the current color attributes, but with the rendition attributes reset (what we refer to as meta attributes). This also includes scroll operations that have to clear the area of the screen that has scrolled into view. The only exception is the _Erase Scrollback_ operation, which needs to reset the buffer with the default attributes. This PR updates all of these cases to apply the correct attributes when scrolling and erasing.

## PR Checklist
* [x] Closes #2553
* [x] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/Terminal) and sign the CLA
* [x] Tests added/passed
* [ ] Requires documentation to be updated
* [ ] I've not really discussed this with core contributors. I'm ready to accept this work might be rejected in favor of a different grand plan. 

## Detailed Description of the Pull Request / Additional comments

My initial plan was to use a special case legacy attribute value to indicate the "standard erase attribute" which could safely be passed through the legacy APIs. But this wouldn't cover the cases that required default attributes to be used. And then with the changes in PR #2668 and #2987, it became clear that our requirements could be better achieved with a couple of new private APIs that wouldn't have to depend on legacy attribute hacks at all.

To that end, I've added the `PrivateFillRegion` and `PrivateScrollRegion` APIs to the `ConGetSet` interface. These are just thin wrappers around the existing `SCREEN_INFORMATION::Write` method and the `ScrollRegion` function respectively, but with a simple boolean parameter to choose between filling with default attributes or the standard erase attributes (i.e the current colors but with meta attributes reset).

With those new APIs in place, I could then update most scroll operations to use `PrivateScrollRegion`, and most erase operations to use `PrivateFillRegion`.

The functions affected by scrolling included:
* `DoSrvPrivateReverseLineFeed` (the RI command)
* `DoSrvPrivateModifyLinesImpl` (the IL and DL commands)
* `AdaptDispatch::_InsertDeleteHelper` (the ICH and DCH commands)
* `AdaptDispatch::_ScrollMovement` (the SU and SD commands)

The functions affected by erasing included:
* `AdaptDispatch::_EraseSingleLineHelper` (the EL command, and most ED variants)
* `AdaptDispatch::EraseCharacters` (the ECH command)

While updating these erase methods, I noticed that both of them also required boundary fixes similar to those in PR #2505 (i.e. the horizontal extent of the erase operation should apply to the full width of the buffer, and not just the current viewport width), so I've addressed that at the same time.

In addition to the changes above, there were also a few special cases, the first being the line feed handling, which required updating in a number of places to use the correct erase attributes:

* `SCREEN_INFORMATION::InitializeCursorRowAttributes` - this is used to initialise the rows that pan into view when the viewport is moved down the buffer.
* `TextBuffer::IncrementCircularBuffer` - this occurs when we scroll passed the very end of the buffer, and a recycled row now needs to be reinitialised.
* `AdjustCursorPosition` - when within margin boundaries, this relies on a couple of direct calls to `ScrollRegion` which needed to be passed the correct fill attributes.

The second special case was the full screen erase sequence (`ESC 2 J`), which is handled separately from the other ED sequences. This required updating the `SCREEN_INFORMATION::VtEraseAll` method to use the standard erase attributes, and also required changes to the horizontal extent of the filled area, since it should have been clearing the full buffer width (the same issue as the other erase operations mentioned above).

Finally, there was the `AdaptDispatch::_EraseScrollback` method, which uses both scroll and fill operations, which could now be handled by the new `PrivateScrollRegion` and `PrivateFillRegion` APIs. But in this case we needed to fill with the default attributes rather than the standard erase attributes. And again this implementation needed some changes to make sure the full width of the active area was retained after the erase, similar to the horizontal boundary issues with the other erase operations.

Once all these changes were made, there were a few areas of the code that could then be simplified quite a bit. The `FillConsoleOutputCharacterW`, `FillConsoleOutputAttribute`, and `ScrollConsoleScreenBufferW` were no longer needed in the `ConGetSet` interface, so all of that code could now be removed. The `_EraseSingleLineDistanceHelper` and `_EraseAreaHelper` methods in the `AdaptDispatch` class were also no longer required and could be removed.

Then there were the hacks to handle legacy default colors in the `FillConsoleOutputAttributeImpl` and `ScrollConsoleScreenBufferWImpl` implementations. Since those hacks were only needed for VT operations, and the VT code no longer calls those methods, there was no longer a need to retain that behaviour (in fact there are probably some edge cases where that behaviour might have been considered a bug when reached via the public console APIs). 

## Validation Steps Performed

For most of the scrolling operations there were already existing tests in place, and those could easily be extended to check that the meta attributes were correctly reset when filling the revealed lines of the scrolling region.

In the screen buffer tests, I made updates of that sort to  the `ScrollOperations` method (handling SU, SD, IL, DL, and RI), the `InsertChars` and `DeleteChars` methods (ICH and DCH), and the `VtNewlinePastViewport` method (LF). I also added a new `VtNewlinePastEndOfBuffer` test to check the case where the line feed causes the viewport to pan past the end of the buffer.

The erase operations, however, were being covered by adapter tests, and those aren't really suited for this kind of functionality (the same sort of issue came up in PR #2505). As a result I've had to reimplement those tests as screen buffer tests.

Most of the erase operations are covered by the `EraseTests` method, except the for the scrollback erase which has a dedicated `EraseScrollbackTests` method. I've also had to replace the `HardReset` adapter test, but that was already mostly covered by the `HardResetBuffer` screen buffer test, which I've now extended slightly (it could do with some more checks, but I think that can wait for a future PR when we're fixing other RIS issues).

### CODE REVIEW

File: src/buffer/out/textBuffer.cpp [hunk 1]
```diff
@@ -549,7 +549,11 @@ bool TextBuffer::IncrementCircularBuffer()
     _renderTarget.TriggerCircling();
 
     // First, clean out the old "first row" as it will become the "last row" of the buffer after the circle is performed.
-    const bool fSuccess = _storage.at(_firstRow).Reset(_currentAttributes);
+    // The VT standard requires that the new row is initialized with
+    // the current background color, but with no meta attributes set.
+    auto fillAttributes = _currentAttributes;
+    fillAttributes.SetMetaAttributes(0);
```
Comment (DHowett-MSFT) [inline src/buffer/out/textBuffer.cpp]: Concerned about this one, actually -- this is used for traditional Windows console apps as well, and I don't know what the expected behavior is for new rows _outside_ of global VT processing mode.

Comment (j4james) [reply src/buffer/out/textBuffer.cpp]: Ah yes. I should have mentioned I did actually have a look into that. I wrote a little test app that set the attributes with `SetConsoleTextAttribute` (using `COMMON_LVB_UNDERSCORE` and `COMMON_LVB_REVERSE_VIDEO`) and then output some text that would cause the screen to scroll.

In both the legacy console and the v2 console, a line feed doesn't actually fill the new row with the current attributes most of the time - only when it reaches the end of the buffer. That's just plain weird, but it seems that's the way it has always been, and that hasn't changed.

In the legacy console the underscore and reverse attributes don't seem to have any effect anywhere, so naturally that means the new rows filled when scrolling past the end of the buffer will also not have an underscore or have their attributes reversed. So in some ways this update more closely matches that legacy behaviour. It is a change from the current v2 behaviour though.

The bottom line is I'd be very surprised if anyone is relying on this behaviour, considering how strange and inconsistent it is. However, I know backward compatibility is important to you, so if you think it's a concern, I can probably add a check that only resets the attributes when in vt mode or something of the sort.

**Comment by j4james** [comment]
There are two other issues I should mention that I'm concerned about in this PR:

1. I noticed that `LockConsole` was being used around most calls `ScrollRegion` (e.g. see [here](https://github.com/microsoft/terminal/blob/94fc40ed31c0de050304248644458b21d0cce2e9/src/host/getset.cpp#L1434-L1441)), so I've followed the same pattern in the new `PrivateScrollRegion` method, as well the in the new `PrivateFillRegion` method around the `SCREEN_INFORMATION::Write` call. However, I've also seen places where I would have expected it to be used, but it wasn't (e.g. see [here](https://github.com/microsoft/terminal/blob/0c8a4df9636982fc92137c47b31877f2c7b5efdb/src/host/_stream.cpp#L145)). So I'm unsure now whether it is really necessary to lock these operations, or whether it's only required for specific cases?

2. I also noticed there's a `NotifyAccessibilityEventing` call in the `FillConsoleOutputAttributeImpl` method to indicate changes to the screen buffer (see [here](https://github.com/microsoft/terminal/blob/4dd9f9c180e00a80b48ebfa8105f99ef2f60c728/src/host/_output.cpp#L240-L248)), but then it's not used in something like the `VtEraseAll` method which also updates the buffer (see [here](https://github.com/microsoft/terminal/blob/4dd9f9c180e00a80b48ebfa8105f99ef2f60c728/src/host/screenInfo.cpp#L2505-L2506)). So again I'm not sure when exactly that is required. For now I haven't called it either the new `PrivateScrollRegion` method or the `PrivateFillRegion` method, but I suspect that may be wrong.

File: src/buffer/out/textBuffer.cpp (hunk 1 above)
Comment (j4james) [reply src/buffer/out/textBuffer.cpp]: I've now made it so this only applies in the specific case of the line feed scrolling when in vt mode. There are probably more situations where it should be applied, but for now this is the most conservative fix for this particular issue.

**Comment by miniksa** [comment]
> There are two other issues I should mention that I'm concerned about in this PR:
> 
> 1. I noticed that `LockConsole` was being used around most calls `ScrollRegion` (e.g. see [here](https://github.com/microsoft/terminal/blob/94fc40ed31c0de050304248644458b21d0cce2e9/src/host/getset.cpp#L1434-L1441)), so I've followed the same pattern in the new `PrivateScrollRegion` method, as well the in the new `PrivateFillRegion` method around the `SCREEN_INFORMATION::Write` call. However, I've also seen places where I would have expected it to be used, but it wasn't (e.g. see [here](https://github.com/microsoft/terminal/blob/0c8a4df9636982fc92137c47b31877f2c7b5efdb/src/host/_stream.cpp#L145)). So I'm unsure now whether it is really necessary to lock these operations, or whether it's only required for specific cases?

It is necessary to lock the scroll operation. We are supposed to lock any operation that moves data around in either the input or output buffers. The fact that it's a global lock pains us, but it's just the way it's been.

The latter case with `AdjustCursorPosition` is probably because it was reasoned out at some previous point in time that that method was never called without already being under `LockConsole`. That might not be strictly true anymore as the code has organically grown, but you've stumbled upon one of the "inherited things that haven't been fully sussed out yet even 5 years later" things. 

Adding a `LockConsole` with unlock inside `AdjustCursorPosition` to cover our butts wouldn't be a bad thing. The lock should be re-entrant. If something starts hanging after you add that, then we've identified a long standing problem.

> 2. I also noticed there's a `NotifyAccessibilityEventing` call in the `FillConsoleOutputAttributeImpl` method to indicate changes to the screen buffer (see [here](https://github.com/microsoft/terminal/blob/4dd9f9c180e00a80b48ebfa8105f99ef2f60c728/src/host/_output.cpp#L240-L248)), but then it's not used in something like the `VtEraseAll` method which also updates the buffer (see [here](https://github.com/microsoft/terminal/blob/4dd9f9c180e00a80b48ebfa8105f99ef2f60c728/src/host/screenInfo.cpp#L2505-L2506)). So again I'm not sure when exactly that is required. For now I haven't called it either the new `PrivateScrollRegion` method or the `PrivateFillRegion` method, but I suspect that may be wrong.

That appears to be another mistake that has slipped through the cracks. We're supposed to be notifying eventing when the contents of the buffer changes in a way that is visible on the screen.

Because that's so prone to error, I've asked @carlos-zamora to start down the path of turning accessibility into a renderer instead of having to separately event it all over the place. The events that accessibility needs to know are pretty much the same things that one needs to know to draw on the screen anyway (and we already have a robust system for knowing and doing that).

However, he's doing it for UIA. Where the `NotifyAccessibilityEventing` method you identified is for MSAA. 

In the short term, it would make sense to propagate the `NotifyAccessibilityEventing` method to the missing one and the new ones. 

In the long term, we should also book moving MSAA to a renderer like @carlos-zamora is doing for UIA. I appear to have mentioned that briefly in #410 as a performance concern as well, but I don't see an issue on the tracker to do it. Maybe @DHowett-MSFT has better GitHub-search-fu than I do.

**Comment by j4james** [comment]
> The latter case with `AdjustCursorPosition` is probably because it was reasoned out at some previous point in time that that method was never called without already being under `LockConsole`. 

I think you're right about that. I added a check to see if that function was ever called without the lock being held and I couldn't get it to fail, so I think it's probably OK. When it's triggered as a result of a write operation it would already have been locked by one of the `WriteConsole*Impl` methods (e.g. see [here](https://github.com/microsoft/terminal/blob/0c8a4df9636982fc92137c47b31877f2c7b5efdb/src/host/_stream.cpp#L1096-L1097)), and for input operations it would likely have come via the `ConsoleWindowProc` method (see [here](https://github.com/microsoft/terminal/blob/56c35945b97ff59621959f621fa9ab5d0ec7e158/src/interactivity/win32/windowproc.cpp#L93-L96)).

But that does suggest that the lock probably shouldn't be needed in any of the private VT methods, since they should already have been locked by the `WriteConsole` call that triggered them. I'm nervous about removing existing locks, though, so I'm leaving those as they are.

> We're supposed to be notifying eventing when the contents of the buffer changes in a way that is visible on the screen.

I've added a `NotifyAccessibilityEventing` to the new `PrivateFillRegion` method, but after a further investigation of the `VtEraseAll`, I think that code is actually fine as it is. The way the screen is cleared is essentially by scrolling down to a clean area of the buffer, and the viewport move already generates a accessibility scroll event for that, so that should be fine.

File: src/host/getset.cpp [hunk 2]
```diff
@@ -2291,3 +2239,99 @@ void DoSrvPrivateMoveToBottom(SCREEN_INFORMATION& screenInfo)
...
+
+        // For most VT erasing operations, the standard requires that the
+        // erased area be filled with the current background color, but with
+        // no additional meta attributes set. For all other cases, we just
+        // fill with the default attributes.
+        auto fillAttrs = TextAttribute{};
+        if (standardFillAttrs)
+        {
+            fillAttrs = screenInfo.GetAttributes();
+            fillAttrs.SetMetaAttributes(0);
```
[the remark this answers is not in the document]
Comment (zadjii-msft) [reply src/host/getset.cpp]: (and below)

File: src/host/screenInfo.cpp [hunk 3]
```diff
@@ -2035,10 +2035,15 @@ const SCREEN_INFORMATION& SCREEN_INFORMATION::GetMainBuffer() const
 
     const FontInfo& existingFont = GetCurrentFont();
 
+    // The buffer needs to be initialized with the standard erase attributes,
+    // i.e. the current background color, but with no meta attributes set.
+    auto initAttributes = GetAttributes();
+    initAttributes.SetMetaAttributes(0);
```
Comment (zadjii-msft) [inline src/host/screenInfo.cpp]: (this guy, and other places in this file, need `ExtendedAttributes` reset too)

File: src/terminal/adapter/adaptDispatch.cpp [hunk 4]
```diff
@@ -1578,64 +1522,47 @@ bool AdaptDispatch::_EraseScrollback()
...
-                    fSuccess = !!_conApi->SetConsoleWindowInfo(true, &srNewViewport);
-
-                    if (fSuccess)
-                    {
-                        // Move the cursor to the same relative location.
-                        const COORD newCursor = { Cursor.X - Screen.Left, Cursor.Y - Screen.Top };
-                        fSuccess = !!_conApi->SetConsoleCursorPosition(newCursor);
-                    }
+                    // Move the cursor to the same relative location.
+                    const COORD newCursor = { Cursor.X, Cursor.Y - Screen.Top };
```
Comment (zadjii-msft) [inline src/terminal/adapter/adaptDispatch.cpp]: Since we're setting the `srNewViewport.Left` to `Screen.Left`, should we use the viewport relative cursor position too?

```suggestion
                    const COORD newCursor = { Cursor.X - Screen.Left, Cursor.Y - Screen.Top };
```

File: src/host/_stream.cpp [hunk 5]
```diff
@@ -72,7 +72,10 @@ using Microsoft::Console::VirtualTerminal::StateMachine;
         }
     }
 
-    const auto bufferAttributes = screenInfo.GetAttributes();
+    // The VT standard requires the lines revealed when scrolling are filled
+    // with the current background color, but with no meta attributes set.
+    auto fillAttributes = screenInfo.GetAttributes();
+    fillAttributes.SetMetaAttributes(0);
```
Comment (zadjii-msft) [inline src/host/_stream.cpp]: (same thing here)

File: src/terminal/adapter/adaptDispatch.cpp (hunk 4 above)
Comment (j4james) [reply src/terminal/adapter/adaptDispatch.cpp]: No, this is deliberate. The viewport has moved up from the original `Screen.Top` to 0, so we need to subtract `Screen.Top` from the original Y coordinate. The viewport's horizontal offset doesn't change though - hence `srNewViewport.Left` is equal to the original `Screen.Left` - and so we don't need to subtract anything from the X coordinate.

File: src/host/getset.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (j4james) [reply src/host/getset.cpp]: No problem. I'll rebase and make those changes everywhere.

In the long term, though, it might be good if we could come up with a plan to merge those methods if possible. It feels a bit error prone having to know to call both of them for something like this.

[the remark this answers is not in the document]
Comment (j4james) [reply src/host/getset.cpp]: Actually it looks like the extended attributes were already there at the time of my last rebase, so I probably won't even need to rebase again.

File: src/host/screenInfo.cpp (hunk 3 above)
Comment (j4james) [reply src/host/screenInfo.cpp]: In case it's not clear, this has been updated (as have all of the other places where the meta attributes are reset). For some reason GitHub hasn't detected that these snippets are all now outdated.

File: src/host/getset.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (zadjii-msft) [reply src/host/getset.cpp]: Yea we should definitely make that simpler in the future. Seeing them all next to each other in this PR like this makes it especially clear that it needs some love.

File: src/terminal/adapter/adaptDispatch.cpp (hunk 4 above)
Comment (zadjii-msft) [reply src/terminal/adapter/adaptDispatch.cpp]: Okay sounds good to me.

**Comment by carlos-zamora** [comment]
@msftbot make sure @miniksa signs off

**Comment by miniksa** [comment]
OK, I intend to read this one after lunch today. Sorry for the delay. I've been swamped with internal deliverables.

File: src/buffer/out/textBuffer.cpp [hunk 6]
```diff
@@ -542,14 +542,22 @@ bool TextBuffer::NewlineCursor()
 // - <none>
```
Comment (miniksa) [inline src/buffer/out/textBuffer.cpp]: Arguments comment not updated.

File: src/host/screenInfo.cpp [hunk 7]
```diff
@@ -2035,10 +2035,16 @@ const SCREEN_INFORMATION& SCREEN_INFORMATION::GetMainBuffer() const
 
     const FontInfo& existingFont = GetCurrentFont();
 
+    // The buffer needs to be initialized with the standard erase attributes,
+    // i.e. the current background color, but with no meta attributes set.
+    auto initAttributes = GetAttributes();
+    initAttributes.SetExtendedAttributes(ExtendedAttributes::Normal);
```
Comment (miniksa) [inline src/host/screenInfo.cpp]: I feel like this pattern (SetExtendedAttributes to Normal, set Meta to 0) is common enough that perhaps there should just be a .SetStandardErase() method on the Attributes that does both of these instead of having this pattern everywhere.

File: src/buffer/out/textBuffer.cpp (hunk 6 above)
Comment (j4james) [reply src/buffer/out/textBuffer.cpp]: Sorry I totally overlooked that. Added in commit 171a4f9dddfef7608fba7f9302050d78e4de66cc

File: src/host/screenInfo.cpp (hunk 7 above)
Comment (j4james) [reply src/host/screenInfo.cpp]: Very happy to do this. Makes things a lot neater. Added in commit 786cc365b1c75ce8192496056c570eef12c39617

File: src/host/getset.cpp [hunk 8]
```diff
@@ -2291,3 +2239,88 @@ void DoSrvPrivateMoveToBottom(SCREEN_INFORMATION& screenInfo)
...
+    try
+    {
+        LockConsole();
+        auto Unlock = wil::scope_exit([&] { UnlockConsole(); });
+
+        // For most VT erasing operations, the standard requires that the
+        // erased area be filled with the current background color, but with
+        // no additional meta attributes set. For all other cases, we just
+        // fill with the default attributes.
+        auto fillAttrs = TextAttribute{};
```
[the remark this answers is not in the document]
Comment (DHowett-MSFT) [reply src/host/getset.cpp]: ARGH i apparently wrote this comment eighteen months ago and never published it
literally don't care now
:smile:

File: src/terminal/adapter/adaptDispatch.cpp [hunk 9]
```diff
@@ -1578,64 +1522,47 @@ bool AdaptDispatch::_EraseScrollback()
...
-            // B. to the right of the viewport.
-
-            // First clear section A
+            // Clear everything after the viewport.
             const DWORD dwTotalAreaBelow = csbiex.dwSize.X * (csbiex.dwSize.Y - sHeight);
             const COORD coordBelowStartPosition = { 0, sHeight };
-            // We don't use the _EraseAreaHelper here because _EraseSingleLineDistanceHelper does it all in one operation
-            fSuccess = _EraseSingleLineDistanceHelper(coordBelowStartPosition, dwTotalAreaBelow, csbiex.wAttributes);
+            // Again we need to use the default attributes, hence standardFillAttrs is false.
+            fSuccess = _conApi->PrivateFillRegion(coordBelowStartPosition, dwTotalAreaBelow, L' ', false);
```
Comment (DHowett-MSFT) [inline src/terminal/adapter/adaptDispatch.cpp]: nit

```suggestion
            fSuccess = _conApi->PrivateFillRegion(coordBelowStartPosition, dwTotalAreaBelow, UNICODE_SPACE, false);
```

if we can see UNICODE_SPACE here?

**Comment by DHowett-MSFT** [comment]
@j4james excellent as usual. There's a couple missing `UNICODE_SPACE`s, but I have my finger on the merge button anyway ;P

**Comment by j4james** [comment]
> There's a couple missing `UNICODE_SPACE`s, but I have my finger on the merge button anyway ;P

Just has a check, and it seems `UNICODE_SPACE` isn't available there anyway. I did actually think I had a reason for using those literals, although it may just have been that I copied them from the previous code.
